entry/src/entry.rs

```diff
@@ -938,6 +954,22 @@ pub fn next_versioned_entry(
     }
 }
 
+pub fn entry_thread_pool_for_tests() -> ThreadPool {
+    rayon::ThreadPoolBuilder::new()
+        .num_threads(4)
```

Comment:
I decided to drop thread count for tests; this this is a reasonable thing to do but calling it out for visibility incase others feel differently.

Also, the changes to the tests are:
- Plumbing the thread pool down
- Adjusting some comments to be on their own lines instead of after - this was to avoid exceeding 100 character count which causes `cargo fmt` to breakdown (and put all on own lines for consistency)

Reply:
Might be nice to comment why we chose 4